addons/knowledge/static/src/webclient/commands/knowledge_providers.js

```diff
@@ -0,0 +1,164 @@
+        <span t-if="props.parentName" t-esc="'— '" class="text-muted small pr-2" />
+        <span t-if="props.parentName" class="o_command_name text-muted small">
+            <t t-foreach="props.splitParentName" t-as="name" t-key="name_index">
+                <b t-if="name_index % 2" t-esc="name"/>
+                <t t-else="" t-esc="name"/>
+            </t>
+        </span>
+    </div>
+`;
+
+// "Not found, create one" command
+class Knowledge404Command extends Component {}
+Knowledge404Command.template = xml`
+    <div class="o_command_hotkey">
+        <span>
+            No Article found. <span class="text-primary">Create "<u t-esc="props.articleName"/>"</span>
+        </span>
+    </div>
+`;
+
+// Advanced search command
+class KnowledgeExtraCommand extends HotkeyCommandItem {}
+KnowledgeExtraCommand.template = xml`
+    <div class="o_command_hotkey">
+        <span>
+            <icon class="fa fa-arrows-alt pr-2" />
+            <t t-esc="props.name" />
+        </span>
+        <span>
+            <t t-foreach="getKeysToPress(props)" t-as="key">
+                <kbd t-esc="key" />
+                <span t-if="!key_last"> + </span>
+            </t>
+        </span>
+    </div>
+`;
+const commandSetupRegistry = registry.category("command_setup");
+commandSetupRegistry.add("?", {
+    emptyMessage: _lt("No article found."),
+    name: _lt("articles"),
+});
+
+const commandProviderRegistry = registry.category("command_provider");
+commandProviderRegistry.add("knowledge", {
+    namespace: "?",
+    async provide(newEnv, options) {
+        // search article's name and parent name
+        const domain = ["|",
+            ["name", "ilike", options.searchValue],
+            ["parent_id.name", "ilike", options.searchValue],
+        ];
+        // retrieve the following fields
+        const fields = ['id', 'name', 'is_user_favourite', 'parent_id', 'icon'];
+        const limit = 10;
+        const articlesData = await Component.env.services.rpc({
+            model: "knowledge.article",
+            method: "search_read",
+            kwargs: {
+                domain,
+                fields,
+                limit,
+            },
+        });
+
+        if (articlesData.length === 0) {
+            // check if user has enough rights to create a new article
+            const canCreate = await Component.env.services.rpc({
+                model: "knowledge.article",
+                method: "check_access_rights",
+                kwargs: {
+                    operation: "create",
+                    raise_exception: false,
+                },
+            });
+            // only display the "create article" command after two characters
+            if (canCreate && options.searchValue.length > 2) {
+                return [{
+                    Component: Knowledge404Command,
+                    async action() {
+                        const article = await Component.env.services.rpc({
```

Comment:
I changes a bit the create, now the controller only returns the id, which is enough to redirect to that record.